main.cpp

```diff
@@ -126,384 +167,529 @@ struct CarWash
     float exportTestTiming = 15.30f;
-    // Frames-per-second when running Game A
     int fpsTestScore = 100;
 
     struct LaptopBottom
     {
-        // Has RGB lighting (bool)
         bool hasRGB = true;
-        // Has a numpad (bool)
         bool hasNumpad = true;
-        // Amount of RAM (int)
         int amountOfRam = 32;
-        // CPU equipped (std::string)
         std::string cpuEquipped = "Ryzen 7 5800H";
-        // GPU equipped (std::string)
         std::string gpuEquipped = "Nvidia RTX3070";
 
-        // Turn laptop on after activating power button
-        bool turnLaptopOn(bool buttonPressed, bool powerPresent); // Returns whether laptop is able to be switched on
-        // Input characters with keyboard
+        bool turnLaptopOn(bool buttonPressed, bool powerPresent);
         void inputCharacter(std::string character, bool keyboardMalfunction = false);
-        // Deliver power using housed battery
-        void powerWithBattery(int batterySize, int powerConsumption, float duration);
+        bool powerWithBattery(int batterySize, int powerConsumption, float duration);
     
     };
 
-    // Launch Game A
     void launchGameA(std::string quality = "Medium");
-    // Export test video
-    int exportTestVideo(int quality = 720, int videoLength = 60); // Return time needed to export video from editing software
-    // Facilitate connection via port to external appliance
+    int exportTestVideo(int quality = 720, int videoLength = 60);
     void connectPortToExternal(std::string portType = "USB", std::string deviceType = "External Drive");
 };
-/*
-Thing 3) Public restroom with management
-5 properties:
-    1) Average number of users per week (float)
-    2) Number of toilets (int)
-    3) Number of urinals (int)
-    4) Number of sinks (int)
-    5) Volume of water used per week (double)
-3 things it can do:
-    1) Unclog toilets
-    2) Clean facilities
-    3) Replenish soap
- */
+
+bool GamingLaptop::LaptopBottom::turnLaptopOn(bool buttonPressed, bool powerPresent)
+{
+    bool powerCommand;
+    if (buttonPressed && powerPresent)
+    {
+        powerCommand = true;
+    }
+    else
+    {
+        powerCommand = false;
+    }
+    return powerCommand;
+}
+
+void GamingLaptop::LaptopBottom::inputCharacter(std::string character, bool malfunction)
+{
+    if (malfunction == false)
+    {
+        std::cout << character << std::endl;
+    }
+    else
+    {
+        std::cout << "Hello your keyboard has broken\n";
+    }
+    return;
+}
+
+bool GamingLaptop::LaptopBottom::powerWithBattery(int capacity, int consumption, float duration)
+{
+    bool powerState;
+    while (capacity > 0)
+    {
+        powerState = true;
+        capacity -= (consumption * duration);
+    }
+    powerState = false;
+    return powerState;
+}
+
+void GamingLaptop::launchGameA(std::string quality)
+{
+    std::cout << "Launching Game A at " << quality << " settings\n";
+    return;
+}
```

Comment:
Functions that don't return anything don't need a `return;` statement
- [ ] revise